src/main/rx/rx.c

```diff
@@ -539,6 +541,52 @@ static void detectAndApplySignalLossBehaviour(void)
 #endif
 }
 
+#define MAX_RC_HISTORY_ITEMS 4
+
+uint16_t rcHistory[MAX_SUPPORTED_RC_CHANNEL_COUNT][MAX_RC_HISTORY_ITEMS];
+
+void updateChannelStability(void)
+{
+    for (uint8_t channelIndex = 0; channelIndex < MAX_SUPPORTED_RC_CHANNEL_COUNT; channelIndex++) {
+
+        // calculate the maximum absolute delta between the new reading each each of the previous readings.
+        uint16_t maximumDelta = 0;
+        for (int8_t historyIndex = -1; historyIndex < MAX_RC_HISTORY_ITEMS - 1; historyIndex++) {
```

Comment:
I would be careful with averaging modes - the filter should catch spikes (almost certainly problem), averaging/summing will reduce spike sensitivity.

Calculating variance/standard deviation may be another option, it is 'natural' way to measure signal stability. It also directly translates to confidence intervals if noise/disturbance is Gaussian.